src/Resources/contao/helper/functions.php

```diff
@@ -15,9 +15,9 @@
  * @param string $strMessage
  * @param string $strLog
  */
-function log_message($strMessage, $strLog='error.log')
```

Comment:
We already talked about this during our first Ahrweiler meeting and AFAIR we came to the conclusion that we can not support `$strLog` anymore and therefore will either have to prepend the message with the value of this or ignore the variable entirely.

For the fact of logging before the container is built. This would AFAIR only affect logging from `config.php` during container compilation, wouldn't it? IMO we should silently ignore the logging request then or use the old functionality as noted in this PR.

I think this pseudocode outlines the best approach:
```php
function log_message($strMessage)
 {
	// $container should be fetched here, if any available.
	if ((func_num_args() === 1) && $container) {
		$container->get('logger')->info($strMessage);
	} else {
		error_log(sprintf("[%s] %s\n", date('d-M-Y H:i:s'), $strMessage), 3, TL_ROOT . '/app/logs/' . $strLog);
		trigger_error(E_USER_DEPRECATED, 'Logging without container and/or to arbitrary destinations is not supported anymore.');
	}
 }
```